app/models/revision.rb

```diff
@@ -75,4 +75,10 @@ def image_revisions_without_lead
   def assets
     image_revisions.flat_map(&:assets) + file_attachment_revisions.map(&:asset)
   end
+
+  def featured_attachments
+    file_attachment_revisions.sort_by do |attachment|
+      featured_attachment_ordering.find_index(attachment.featured_attachment_id)
```

Comment:
I think this is liable to crash if we have a situation where one of the attachments doesn't exist in the featured_attachment_ordering array (it'd crash when it compares an integer with a nil).

Might need this to fall back to Float::INFINITY so any missing items end up at the end of the list.

Reply:
That's an interesting point. It looks like `sort_by` copes with this, which is why https://github.com/alphagov/content-publisher/pull/1918/files#diff-da3c22393aa90b5ca4f6c96d0459b297R3 is passing.

Reply:
this would cope with every item not being in the list (as all nils would be compared to nil) but  would error once you have a mixture (int compared to nil)

Reply:
Ah, I understand. Good spot - I've added an extra test and updated 👍